lib/neo4j/active_node/persistence.rb

```diff
@@ -128,15 +128,19 @@ def create!(*args)
         end
       end
 
-      def merge(attributes)
-        neo4j_session.query.merge(n: {self.mapped_label_names => attributes})
-          .on_create_set(n: on_create_props(attributes))
-          .on_match_set(n: on_match_props)
+      def merge(match_attributes, optional_attrs = {})
+        optional_attrs.default = {}
+        on_create_set, on_match_set, set_attrs = optional_attrs.values_at :on_create, :on_match, :set
```

Comment:
To keep things consistent I would call the attributes `on_create_attrs`, `on_match_attrs`, and `set_attrs`.

Also, we should have something like:

```ruby
invalid_options = optional_attrs.keys - [:on_create, :on_match, :set]
fail ArgumentError, "Invalid options: #{invalid_options.inspect}"
```

Honestly this happens so much we should probably have a helper method.  ... a couple of minutes pass ... looks like ActiveSupport already does that!

http://apidock.com/rails/Hash/assert_valid_keys

We're including `ActiveSupport` in the gem so you should just need to:

```ruby
require 'active_support/core_ext/hash/keys'
```